src/NBitcoin/ConsensusOptions.cs

```diff
@@ -28,24 +36,127 @@ public class ConsensusOptions
-            this.SerializeTransactionNoWitness = 0x40000000;
             this.MaxStandardVersion = 2;
             this.MaxStandardTxWeight = 400000;
             this.MaxBlockBaseSize = 1000000;
             this.MaxBlockSigopsCost = 80000;
         }
+
+        /// <summary>
+        /// Initializes all values. Used by networks that use block weight rules.
+        /// </summary>
+        public ConsensusOptions(
+            uint maxBlockBaseSize,
+            uint maxBlockWeight,
+            uint maxBlockSerializedSize,
+            int witnessScaleFactor,
+            int maxStandardVersion,
+            int maxStandardTxWeight,
+            int maxBlockSigopsCost)
+        {
+            this.MaxBlockBaseSize = maxBlockBaseSize;
+            this.MaxBlockWeight = maxBlockWeight;
+            this.MaxBlockSerializedSize = maxBlockSerializedSize;
+            this.WitnessScaleFactor = witnessScaleFactor;
+            this.MaxStandardVersion = maxStandardVersion;
+            this.MaxStandardTxWeight = maxStandardTxWeight;
+            this.MaxBlockSigopsCost = maxBlockSigopsCost;
+        }
+
+        /// <summary>
+        /// Initializes values for networks that use block size rules.
+        /// </summary>
+        public ConsensusOptions(
+            uint maxBlockBaseSize,
+            int maxStandardVersion,
+            int maxStandardTxWeight,
+            int maxBlockSigopsCost)
+        {
+            this.MaxBlockBaseSize = maxBlockBaseSize;
+
+            // Having witnessScale = 1 and setting the weight to be the same value as the base size
+            // will result in all checks comparing size in bytes.
+            this.MaxBlockWeight = maxBlockBaseSize;
+            this.MaxBlockSerializedSize = maxBlockBaseSize;
+            this.WitnessScaleFactor = 1;
+
+            this.MaxStandardVersion = maxStandardVersion;
+            this.MaxStandardTxWeight = maxStandardTxWeight;
+            this.MaxBlockSigopsCost = maxBlockSigopsCost;
+        }
+    }
+
+    /// <summary>
+    /// Extension to ConsensusOptions for PoS-related parameters.
+    /// 
+    /// TODO: When moving rules to be part of consensus for network, move this class to the appropriate project too.
+    /// Doesn't make much sense for it to be in NBitcoin.
+    /// </summary>
+    public class PosConsensusOptions : ConsensusOptions
+    {
+        // TODO: Remove these 2 properties and set CointakeMinConfirmation in Network.
+
+        /// <summary>Coinstake minimal confirmations softfork activation height for the mainnet.</summary>
+        public const int CoinstakeMinConfirmationActivationHeightMainnet = 1005000;
+
+        /// <summary>Coinstake minimal confirmations softfork activation height for the testnet.</summary>
+        public const int CoinstakeMinConfirmationActivationHeightTestnet = 436000;
+
+        /// <summary>
+        /// Initializes the default values.
+        /// </summary>
+        public PosConsensusOptions()
+        {
```

Comment:
Should this be setting some values?

Reply:
Nope - uses empty constructor in base, which sets everything. 

Nothing actually gets set in PoSConsensusOptions at the moment, it's just used for that `GetStakeMinConfirmations`

Reply:
Ok its good for now but we can do better.